sdjournal/journal_test.go

```diff
@@ -150,3 +151,47 @@ func TestJournalCursorGetSeekAndTest(t *testing.T) {
 		t.Fatalf("Error testing cursor to journal: %s", err)
 	}
 }
+
+func TestJournalGetEntry(t *testing.T) {
+	j, err := NewJournal()
+	if err != nil {
+		t.Fatalf("Error opening journal: %s", err)
+	}
+
+	if j == nil {
+		t.Fatal("Got a nil journal")
+	}
+
+	m := Match{Field: "_CMDLINE", Value: strings.Join(os.Args, " ")}
+	j.AddMatch(m.String())
+
+	want := fmt.Sprintf("test message %s", time.Now())
+	err = journal.Print(journal.PriInfo, want)
+	if err != nil {
+		t.Fatalf("Error writing to journal: %s", err)
+	}
+
+	r := j.Wait(time.Duration(1) * time.Second)
+	if r < 0 {
+		t.Fatalf("Error waiting to journal")
+	}
+
+	n, err := j.Next()
+	if err != nil {
+		t.Fatalf("Error reading to journal: %s", err)
+	}
+
+	if n == 0 {
+		t.Fatalf("Error reading to journal: %s", io.EOF)
+	}
+
+	entry, err := j.GetEntry()
+	if err != nil {
+		t.Fatalf("Error getting the entry to journal: %s", err)
+	}
+
+	got := entry.Fields["MESSAGE"]
+	if got != want {
+		t.Fatalf("Bad result for entry.Fields[\"MESSAGE\"]: got %s, want %s", got, want)
+	}
+}
```

Comment:
This test is fragile, it will easily pick up messages from other tests and get confused. I think you should instead filter on something a bit more specific (for example introducing a custom field key with a random entry, and matching on that).